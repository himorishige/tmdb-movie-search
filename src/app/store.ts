import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import userReducer from 'src/features/user/userSlice';
import { setupListeners } from '@reduxjs/toolkit/query';
import { movieApi } from 'src/features/movie/movieQuery';

export const store = configureStore({
  reducer: {
    user: userReducer,
    [movieApi.reducerPath]: movieApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(movieApi.middleware),
});

setupListeners(store.dispatch);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
