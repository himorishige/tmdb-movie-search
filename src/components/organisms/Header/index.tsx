import { memo } from 'react';
import { Box, Heading, Flex, Icon, Container, Text } from '@chakra-ui/react';
import { MdFavorite } from 'react-icons/md';
import { Link } from 'react-router-dom';
import SearchArea from '../SearchArea';
import { useAppSelector } from 'src/app/hooks';
import { selectFavoritesIds } from 'src/features/user/userSlice';

const Header: React.VFC = memo(() => {
  const favorites = useAppSelector(selectFavoritesIds);

  return (
    <Box pt="1rem" pb="1.5rem" bgGradient="linear(blue.900 0%, blue.700 100%)">
      <Container maxW="container.lg">
        <Flex direction="column" w="100%">
          <Flex alignItems="center" justifyContent="space-between">
            <Heading
              as={Link}
              to="/"
              color="white"
              _hover={{ opacity: '0.8' }}
              transition="opacity 0.3s"
            >
              TMDb Search
            </Heading>
            <Box pt={2}>
              {favorites.length ? (
                <Box position="relative">
                  <Link to="/mypage">
                    <Icon
                      data-testid="heart-icon"
                      as={MdFavorite}
                      w={8}
                      h={8}
                      color="red.400"
                      _hover={{ color: 'red.200' }}
                      transition="0.3s"
                    />
                  </Link>
                  <Flex
                    position="absolute"
                    p="1"
                    color="white"
                    top="-22px"
                    right="-4px"
                    alignItems="center"
                  >
                    <Text data-testid="favorites-count">{favorites.length.toString()}</Text>
                  </Flex>
                </Box>
              ) : (
                <Link to="/mypage">
                  <Icon data-testid="heart-icon" as={MdFavorite} w={8} h={8} color="gray.400" />
                </Link>
              )}
            </Box>
          </Flex>
          <Box pt="1rem">
            <SearchArea />
          </Box>
        </Flex>
      </Container>
    </Box>
  );
});

export default Header;
