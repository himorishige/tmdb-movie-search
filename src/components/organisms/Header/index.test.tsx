import renderer from 'react-test-renderer';
import Header from '.';
import { store } from 'src/app/store';
import { Provider } from 'react-redux';
import { HelmetProvider } from 'react-helmet-async';
import { ChakraProvider } from '@chakra-ui/react';
import { BrowserRouter } from 'react-router-dom';

describe('Organisms/Header', () => {
  test('正しくレンダリングされている', () => {
    const header = renderer
      .create(
        <HelmetProvider>
          <Provider store={store}>
            <ChakraProvider>
              <BrowserRouter>
                <Header />
              </BrowserRouter>
            </ChakraProvider>
          </Provider>
        </HelmetProvider>,
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
