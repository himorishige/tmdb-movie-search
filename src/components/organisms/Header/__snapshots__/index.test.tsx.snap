// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Header 正しくレンダリングされている 1`] = `
Array [
  <div
    className="css-1tp0brz"
  >
    <div
      className="chakra-container css-1xo1xlh"
    >
      <div
        className="css-1ofqig9"
      >
        <div
          className="css-1lekzkb"
        >
          <a
            className="chakra-heading css-1s7rbqn"
            href="/"
            onClick={[Function]}
          >
            TMDb Search
          </a>
          <div
            className="css-wt5l11"
          >
            <a
              href="/mypage"
              onClick={[Function]}
            >
              <svg
                className="chakra-icon css-19aatzn"
                data-testid="heart-icon"
                fill="currentColor"
                focusable={false}
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          className="css-17bmav7"
        >
          <div
            className="css-0"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="css-k008qs"
              >
                <input
                  className="chakra-input css-ykzfd5"
                  data-testid="input"
                  defaultValue=""
                  disabled={false}
                  enterKeyHint="search"
                  name="keyword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                />
                <button
                  className="chakra-button css-2b4k24"
                  disabled={false}
                  type="submit"
                >
                  検索
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;
